### ISSUE TITLE
disable GoBack

### ISSUE THREAD

**Issue by markFleischmann** [MicrosoftEdge/WebView2Feedback#758]
**Is your feature request related to a problem? Please describe.**
I need a way to prevent the user from going back. I see the 'CanGoBack' Property but it is read only. In my app, I don't provide a back button but you can still use the mouse back button or ALT + Left arrow, etc.

**Describe the solution you'd like and alternatives you've considered**
I tried to prevent this in the "CoreWebView2.NavigationStarting" event handler but I am not able to tell if the user is going back or just navigating to a new page. In the "NavigationStarting" event, if you could provide a isGoingBack/isGoingForward flag in the CoreWebView2NavigationStartingEventArgs, that would solve the issue since you could disable going back/forward by setting Cancel to true.

**Comment by Symbai** [comment]
Cant you just save the `webView.CoreWebView2.Source` on `NavigationCompleted` event and compare it against the page the user wants to navigate to in the `NavigationStarting` event?

**Comment by markFleischmann** [comment]
@Symbai Thanks for the suggestion but I was hoping to disable going back specifically. If you were to navigate to one website and then navigate to a second website and then navigate back to the first website (not by using goBack) your solution would prevent this and I would like to not prevent that.

**Comment by hunkydoryrepair** [comment]
Yes, back button can lead to all kinds of problems. Can have unexpected results with single page apps or even introduce security issues on a shared computer (say, a public kiosk, point of sale, etc.).

**Comment by Kitiara** [comment]
> **Is your feature request related to a problem? Please describe.**
> I need a way to prevent the user from going back. I see the 'CanGoBack' Property but it is read only. In my app, I don't provide a back button but you can still use the mouse back button or ALT + Left arrow, etc.

You disable that keypress by adding [add_AcceleratorKeyPressed event](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2controller?view=webview2-1.0.705.50#add_acceleratorkeypressed). A working example:

        RETURN_IF_FAILED(m_contentController->add_AcceleratorKeyPressed(Callback<ICoreWebView2AcceleratorKeyPressedEventHandler>(
            [this](ICoreWebView2Controller* sender, ICoreWebView2AcceleratorKeyPressedEventArgs* args) -> HRESULT
        {
                COREWEBVIEW2_KEY_EVENT_KIND kind;
                RETURN_IF_FAILED(args->get_KeyEventKind(&kind));
                if (kind == COREWEBVIEW2_KEY_EVENT_KIND_KEY_DOWN || kind == COREWEBVIEW2_KEY_EVENT_KIND_SYSTEM_KEY_DOWN)
                {
                    UINT key;
                    RETURN_IF_FAILED(args->get_VirtualKey(&key));
                    if (key == 0x25) // Left Arrow
                    {
                        COREWEBVIEW2_PHYSICAL_KEY_STATUS status;
                        RETURN_IF_FAILED(args->get_PhysicalKeyStatus(&status));
                        if ((GetKeyState(VK_MENU) & 0x8000)) // ALT + Left Arrow
                            RETURN_IF_FAILED(args->put_Handled(TRUE));
                    }
                }

            return S_OK;
        }).Get(), &m_acceleratorKeyPressedToken));

**Comment by cbra-caa** [comment]
@Kitiara This still doesn't solve for the problem for macro buttons, like you have on certain Mouse or Keyboards.

We would also like a way to determine/block this as we are mostly using it as a mail viewer. And in that case move back and forward between the mails does not make a lot of sense, unless it is handled by the surrounding view as well.

**Comment by fvanheeswijk** [comment]
Is there any news on this issue?

**Comment by champnic** [comment]
Thanks for the callout, looks like we missed this one. We already have an item tracking #1045 on our backlog which is quite similar, so I've linked this to that scenario.

**Comment by kirsan31** [comment]
+1 here. It's very needed functionality - completely disable possibility of Back/Forward navigating.
Navigating Back/Forward can occur:

1. From context menu. 
  Workaround: only completely disable context menu `CoreWebView2Settings.AreDefaultContextMenusEnabled`. Need https://github.com/MicrosoftEdge/WebView2Feedback/issues/80 for more proper solution.
2. From ALT + Left/Right arrow.
  Workaround:  [add_AcceleratorKeyPressed event](https://github.com/MicrosoftEdge/WebView2Feedback/issues/758#issuecomment-770328046). Not working on WinForms https://github.com/MicrosoftEdge/WebView2Feedback/issues/721 
3. Mouse XButton1 and XButton2.
  No workaround. https://github.com/MicrosoftEdge/WebView2Feedback/issues/1045

Overall I totally agree with author, that his suggestion will be the best solution:

> I tried to prevent this in the `CoreWebView2.NavigationStarting` event handler but I am not able to tell if the user is going back or just navigating to a new page. **In the `NavigationStarting` event, if you could provide a `isGoingBack/isGoingForward` flag in the `CoreWebView2NavigationStartingEventArgs`, that would solve the issue since you could disable going back/forward by setting Cancel to true.**

**Comment by pavankumar133** [comment]
Any probable date when this feature will be provided to disable goback

**Comment by champnic** [comment]
@pavankumar133 This is assigned to a dev and I believe the design work is complete, but I don't yet know or have a good estimate of when the work will be done. Probably over the next couple of months. Thanks!

**Comment by zzultc** [comment]
hi there,
I have a method, to clear the history-record when NavigationCompleted is called.

CoreWebView2BrowsingDataKinds dataKinds = CoreWebView2BrowsingDataKinds.BrowsingHistory;
 await webView2.CoreWebView2.Profile.ClearBrowsingDataAsync(dataKinds);

and it works.
hope it can help you.

**Comment by fvanheeswijk** [comment]
Is there any news on this?
